packages/jest-config/src/index.js

```diff
@@ -67,22 +67,31 @@ const getConfigs = (
       coverageThreshold: options.coverageThreshold,
       expand: options.expand,
       forceExit: options.forceExit,
+      json: options.json,
+      lastCommit: options.lastCommit,
+      listTests: options.listTests,
       logHeapUsage: options.logHeapUsage,
       mapCoverage: options.mapCoverage,
+      maxWorkers: options.maxWorkers,
+      noSCM: undefined,
```

Comment:
this one is complicated, because we only know its value after we searched for tests (`jest-changed-files`).
currently in Jest master we modify `globalConfig` after we get the result and i think this is not a good pattern.
I think we should split `runJest` function into two and run them sequentially:

```js
const globalConfig = readConfig(...);
const {noSCM, tests} = findTests(globalConfig);
globalConfig.noSCM = noSCM;
globalConfig.somethingElse = ... // there's a few modifications that we're already doing in runJest

Object.freeze(globalConfig); // at this point config is immutable. no hacks
runJest(globalConfig);
```